ykrt/src/mt.rs

```diff
@@ -741,7 +753,36 @@ impl MT {
                 }
                 None => {
                     if is_tracing {
-                        // This thread is tracing something, so bail out as quickly as possible
+                        let hl_ptr = match loc.inc_count() {
+                            Some(count) => {
+                                let hl = HotLocation {
+                                    kind: HotLocationKind::Counting(count),
+                                    tracecompilation_errors: 0,
+                                };
+                                loc.count_to_hot_location(count, hl)
+                                    .map(|x| Arc::as_ptr(&x))
+                            }
+                            None => loc.hot_location().map(|x| x as *const Mutex<HotLocation>),
+                        };
+                        if let Some(hl_ptr) = hl_ptr {
+                            let MTThreadState::Tracing {
+                                frameaddr: tracing_frameaddr,
+                                seen_hls,
+                                ..
+                            } = mtt.peek_mut_tstate()
+                            else {
+                                panic!()
+                            };
+                            if frameaddr != *tracing_frameaddr {
```

Comment:
Is this bit necessary, or something unrelated we are sneaking in?

Reply:
[i.e. it should always have been present]

Reply:
Do you want to keep it?

Reply:
It should always have been in, and it directly affects this code, so yes.

Reply:
sorry, we raced. All ok.